internal/pkg/consensus/expected.go

```diff
@@ -202,7 +203,9 @@ func (c *Expected) validateMining(ctx context.Context,
 			return errors.Wrapf(err, "failed to convert address, %s, to a signing address", workerAddr.String())
 		}
 		// Validate block signature
-		if err := crypto.ValidateSignature(blk.SignatureData(), workerSignerAddr, blk.BlockSig); err != nil {
+		sigData := blk.SignatureData()
+		fmt.Printf("sigData: %x\n", sigData)
+		if err := crypto.ValidateSignature(blk.SignatureData(), workerSignerAddr, *blk.BlockSig); err != nil {
```

Comment:
This will panic if `blk.BlockSig` is nil. Are we guaranteed it won't be at this point? Some validation would make me feel better.

Reply:
That's why I added [this](https://github.com/filecoin-project/go-filecoin/pull/3897/files#diff-339af0e38c9bd41224f91100b0f74b51R116)

We can be paranoid and do it here if you think that's a good idea.  However we are already in such a bad state that panicing the node is probably the right response if we are doing this validation on a block that fails syntactic validation

Reply:
I'm just going to be paranoid and check for nil here too